blog: v6.4.1 announcement (#6128)
simplex-chat/simplex-chat: blog/20250729-simplex-chat-v6-4-1-welcome-contacts-protect-groups-app-security.md

# SimpleX Chat v6.4.1: welcome your contacts, review members to protect groups, and more.

**Published:** Jul 29, 2025

**What's new in v6.4.1**:

> - [welcome your contacts](#welcome-your-contacts) with your bio and welcome message.

Reviewer: welcome your contacts with bio sounds strange, why not this:
```suggestion
- [welcome your contacts](#welcome-your-contacts): set profile bio and welcome message.
```
as in what's new

> - [protect your groups](#protect-your-groups) and communities from spam and abuse:

Reviewer: ```suggestion
- [protect your groups](#protect-your-groups) from spam and abuse:
```

  - review new members,
  - moderator role to delegate message moderation to trusted members,
  - receive direct feedback from your group members.
- [other UX improvements](#other-improvements).

Also, we added 4 new interface language to the Android and desktop apps: Catalan, Indonesian, Romanian and Vietnamese.

Huge thanks to our users who [contributed translations](https://github.com/simplex-chat/simplex-chat#help-translating-simplex-chat).

## What's new in v6.4.1

### Welcome your contacts

TODO image

We wrote before about the new user experience of connecting with people - it was released in v6.4.

> To maintain backward compatibility, with v6.4 your short link included only your profile name and chat preferences.

Reviewer: I think this is too modest for announcement of that UX rework. Previous post was going to be technical explanation, and this was going to have proper announcement, but now first thing it does is refers to previous post. At least it could be rephrased, smth like this:

"
New user experience of connecting with people is finally here [maybe less cheesy] / is released in full capacity. It \<does something something\>...      [so - first big announcement]

v6.4.1 enables images, bio ...      [second is what changed]

See technical explanation of how it works in previous post \<link\>.       [last - refers to old post]
"

Reviewer: also, there's no link to the previous post

The links and addresses in this version, your contacts can see before connecting:
- your profile name and bio,
- your profile image,

> - for SimpleX addresses and group links - welcome message.

Reviewer: doesn't make sense grammatically

> It allows you to start chatting with your contacts and group members even before they connect.

Reviewer: group members?

> For previously created addresses you will have an option to upgrade. It will make them short, and include profiles into link data. Old long addresses will continue to work, so you won't lose any contacts during the upgrade.

Reviewer: ```suggestion
For previously created addresses you have an option to upgrade. It will make them short, and include profiles into link data. Old long addresses will continue to work, so you won't lose any contacts during the upgrade.
```

And as before, it is as secure from servers - servers cannot see you profiles, unless they have the link, and cannot modify them even if they do have the link.

Read more in the mentioned post about security model of new SimpleX links and addresses.

> And now you can share your short address in your social media profiles - it's less than 80 characters.

Reviewer: awkward placing.. probably better to put it in "big announcement" part

### Protect your groups

TODO images

**Review new members**

Since v6.4 there are some major improvements in your ability to protect your group from spam and abuse.

You can enable an option to "review" all new group members. It is also commonly called "knocking". It allows you to:
- ask prospective members any questions,
- explain the group rules,
- make sure their profile is appropriate for the group,
- decide whether to allow them joining the group, and whether they should be able to send messages in the group.

> Some groups for small communities may enable "review" permanently, while larger public groups may enable it temporarily during spam/troll attacks.

Reviewer: ```suggestion
Some groups for small communities may enable member review permanently, while larger public groups may enable it temporarily during spam/troll attacks.
```

**New role for group moderators**

In addition to that, there is a new group role - moderator.

This role allows:
- to approve members in review,
- moderate messages,
- block members for all.

> Unlike Admins, moderators can't add new members or permanently remove members from the group. This allows you to delegate group moderation to your community members without risking that they may disrupt the group.

Reviewer: why capital, it's not capital in UI
```suggestion
Unlike admins, moderators can't add new members or permanently remove members from the group. This allows you to delegate group moderation to your community members without risking that they may disrupt the group.
```

**Receive direct feedback from group members**

Your group members now can send group admins messages. Each conversation with a group member is a mini-group where all group owners, admins and moderators can talk to a member. Reports that members can send since v... are also added to chat with member, allowing you to discuss the report.

### Other improvements

**Enable disappearing messages for new contacts**

Now you can enable disappearing messages for all new contacts automatically. Tap your profile image in the corner, then tap Chat preferences and set time for messages to disappear.

**Improved message delivery**

We improved networking layer by increasing request timeouts for all background requests. It substantially reduces traffic on slow networks.

### Improved app integrity

**Supply chain security**

> The app security depends on security of its components, and many of these components are created by third parties. In this version we changed the build process to make changes of these components controlled by us:

Reviewer: changed to make changes - what does it mean

- all 3rd party GitHub actions used during the build are now moved to [the forks we control](https://github.com/simplex-chat?q=action&type=fork&sort=name) - it prevents supply chain attacks via build actions.
- we now build VLC library for all platforms that use it from the source code ourselves, in [this repository](https://github.com/simplex-chat/vlc).
- SQLCipher and [Haskell dependencies](https://github.com/simplex-chat/simplex-chat/blob/stable/docs/dependencies/HASKELL.md) versions were already "locked" prior to this version.

**Automatic virus scanning**

We now run automatic daily virus scanning of all apps released via GitHub using [VirusTotal.com](https://www.virustotal.com/).

You can see the scan results [here](https://github.com/simplex-chat/simplex-virutstotal-scan).

**Reproducible builds**

In addition to [server builds](https://github.com/simplex-chat/simplexmq/releases/tag/v6.4.1) that were reproducible since v6.3, the builds of Linux CLI and desktop apps are now reproducible too. You can build Linux apps from source using this script.

> *Please note*: there are some system level dependencies that would change the build when they are updated, which happens rarely.

Reviewer: not trying to nitpick here, it's just it sounds a bit like "other fixes"
